src/main/java/ru/yandex/practicum/filmorate/storage/InMemoryFilmStorage.java

```diff
@@ -0,0 +1,83 @@
+package ru.yandex.practicum.filmorate.storage;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import ru.yandex.practicum.filmorate.exceptions.ObjectNotExistException;
+import ru.yandex.practicum.filmorate.exceptions.ValidationException;
+import ru.yandex.practicum.filmorate.interfaces.FilmStorage;
+import ru.yandex.practicum.filmorate.model.Film;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+@Slf4j
+public class InMemoryFilmStorage implements FilmStorage {
+    private final Map<Long, Film> films = new HashMap<>();
+    private int id = 0;
+
+    @Override
+    public Film addFilm(Film film) {
+        if (!isFilmInfoValid(film)) {
+            log.debug("Валидация при сохранении фильма не пройдена!");
+            throw new ValidationException("Информация о фильме не проходит условия валидации. Фильм не добавлен!");
+        }
+        film.setId(++id);
+        films.put(film.getId(), film);
+        log.debug("Добавлен новый фильм: " + film);
+        return film;
+    }
+
+    @Override
+    public Film updateFilm(Film film) {
+        if (!isFilmInfoValid(film)) {
+            log.debug("Валидация при обновлении фильма не пройдена!");
+            throw new ValidationException("Информация о фильме не проходит условия валидации. Фильм не добавлен!");
+        } else if (!films.containsKey(film.getId())) {
+            log.debug("Попытка обновления несуществующего фильма!");
+            throw new ObjectNotExistException("Попытка обновления несуществующего фильма!");
+        }
+        films.put(film.getId(), film);
+        log.debug("Обновлен фильм: " + film);
+        return film;
+    }
+
+    @Override
+    public Film deleteFilm(Film film) {
+        films.remove(film.getId());
+        log.debug("Удалён фильм: " + film);
+        return film;
+    }
+
+    public Film getFilmById(Long filmId) {
+        return films.get(filmId);
+    }
+
+    @Override
+    public List<Film> getFilms() {
+        return new ArrayList<>(films.values());
+    }
+
+    public void addLike(Long userId, Long filmId) {
+        films.get(filmId).addLike(userId);
+    }
+
+    public void deleteLike(Long userId, Long filmId) {
+        Film film = films.get(filmId);
```

Comment:
если фильма нет в хранилище (в мапе), то переменная остается null, а к null нельзя вызвать метод getUsersLiked(), который встречается в следующем ифе. Именно в этом моменте вываливается исключение:
```
{
    "error": "Возникло исключение",
    "description": "Cannot invoke \"ru.yandex.practicum.filmorate.model.Film.getUsersLiked()\" because \"film\" is null"
}
```
это дефолтное Throwable исключение. Предлагаю вынести проверку наличия в хранилище фильма вынести на уровень сервиса, там же в случае его отсутствия бросать кастомное рантайм исключение. Это будет удобно на следующем спринте, не придется много переделывать))